server/punishments.ts

```diff
@@ -380,22 +339,23 @@ const Punishments = new (class {
 	async loadBanlist() {
 		const data = await FS('config/ipbans.txt').readIfExists();
 		if (!data) return;
-		let rangebans = [];
+		const rangebans = [];
 		for (const row of data.split("\n")) {
 			const ip = row.split('#')[0].trim();
 			if (!ip) continue;
 			if (ip.includes('/')) {
 				rangebans.push(ip);
 			} else if (!Punishments.ips.has(ip)) {
-				Punishments.ips.set(ip, ['BAN', '#ipban', Infinity, '']);
+				Punishments.ips.set(ip, ['BAN', '#ipban' as ID, Infinity, '']);
```

Comment:
Why are we casting this to `ID`? It's not an ID.

If the `Punishment` array takes non-ID string, you should probably just change the type itself, instead of abusing type casting.

Reply:
(it's being asserted as that for ease of use since all the relevant code checks for it, left a comment to that extent)

Reply:
We could also type Punishments#ips as `PunishmentMap<string, punishment with second key string>` and the userid punishments as `PunishmentMap<ID, ID>` (and the same for nested maps), but that turns the earlier code into ] `>><><<>` soup, which I'm not sure is an improvement overall.